README.md

```diff
@@ -1,2 +1,61 @@
 # iatlas-scripts
-A home for experimental scripts used for launching Nextflow Tower workflows for iAtlas
+A home for experimental scripts used for launching Nextflow Tower workflows for iAtlas Data Processing
+
+## Setup
+
+These instructions assume that you already have Python, a Python version manager (`pyenv`), and `pipenv` installed.
+
+### Python Environment
+
+Set up your Python environment for using these scripts by ruinning
+```
+pipenv install
+pipenv shell
+```
+
+### Environment Variables
+
+In order for the scripts leveraging `py-orca` to connect to Nextflow Tower, you must configure a `NEXTFLOWTOWER_CONNECTION_URI` in your local environment. To do so, you will need a Nextflow Tower token with access to the workspace that you wish to use (`Sage-Bionetworks/iatlas-project` in this case). You can then copy `.env.example` into a local `.env` file, replace `<tower-access-token>`, and run `source .env` in your terminal.
+
+## Immune Subtype Clustering
+
+Steps to run the workflow for Immune Subtype Clustering:
+1. Upload all sample files to a folder on Synapse. Make sure that the only `.tsv` files in the folder are the ones that you want to be processed.
+2. Prepare the master data sheet by executing `immune_subtype_clustering/prepare_data_sheet.py` with three arguments:
+    - `parent`: Synapse ID of the folder where your data files are
+    - `export_name`: Name that you want the master data file to be exported to
+    - `upload_location`: Synapse ID of the folder that you want to upload the master data file to
+```
+python immune_subtype_clustering/prepare_data_sheet.py <parent> <export_name> <upload_location>
+```
+3. Create and store your CWL `.json` configuration file in the same location in Synapse as the data file produced by the previous step.
+Example `.json` file:
+``` immune_subtype_clustering_input.json
+{
+    "input_file": {
+        "path": <export_name>,
+        "class": "File"
+    },
+    "input_gene_column": "gene"
+}
+```
+4. Create and store your `nf-synstage` input `.csv` file in an S3 bucket accessible to the Nextflow Tower workspace (`s3://iatlas-project-tower-bucket` or `s3://iatlas-project-tower-scratch`).
```

Comment:
For future iteration, should `prepare_data_sheet.py` also create the sample sheet?  Right now you have a manual step of creating the sample sheet which is probably fine.

If you are doing this step manually, I recommend exploring Nextflow Tower Datasets in the future, but this is good as is.